vpr/src/noc/read_xml_noc_traffic_flows_file.h

```diff
@@ -0,0 +1,190 @@
+ * @param loc_data Contains location data about the current line in the xml file
+ * @param cluster_ctx Global variable that contains clustering information. Used
+ *                    to get information about the router blocks int he design.
+ * @param noc_ctx  Global variable that contains NoC information. Used to access
+ *                 the NocTrafficFlows class and store current flow.
+ * @param noc_router_tile_type The physical type of a Noc router tile in the
+ *                             FPGA.
+ */
+void process_single_flow(pugi::xml_node single_flow_tag, const pugiutil::loc_data& loc_data, const ClusteringContext& cluster_ctx, NocContext& noc_ctx, t_physical_tile_type_ptr noc_router_tile_type);
+
+/**
+ * @brief Checks to see that the two router module names provided in the 
+ *        traffic flow description are not empty and they dont have the
+ *        same names. THe two routers cant be the exact same since a router
+ *        cannot communicate with itself.
+ * 
+ * @param source_router_name A string value of the source router module name
+ * @param sink_router_name A string value of the sink router
+ *                                module name
+ * @param single_flow_tag A xml tag that contains the traffic flow information
+ * @param loc_data Contains location data about the current line in the xml file
+ */
+void verify_traffic_flow_router_modules(std::string source_router_name, std::string sink_router_name, pugi::xml_node single_flow_tag, const pugiutil::loc_data& loc_data);
+
+/**
+ * @brief Ensures that the a given traffic flows data transmission size and
+ *        latency constraints are not negative values.
+ * 
+ * @param traffic_flow_bandwidth The transmission size betwee the two routers
+ *                               in the traffic flow.
+ * @param max_traffic_flow_latency The allowable latency for the data
+ *                                 transmission between the two routers in the
+ *                                 traffic flow.
+ * @param single_flow_tag A xml tag that contains the traffic flow information
+ * @param loc_data Contains location data about the current line in the xml file
+ */
+void verify_traffic_flow_properties(double traffic_flow_bandwidth, double max_traffic_flow_latency, pugi::xml_node single_flow_tag, const pugiutil::loc_data& loc_data);
+
+/**
+ * @brief Given a router module name in the design, retrieve the
+ *        equivalent clustered router block identifier, which is
+ *        a ClusterBlockId.
+ * 
+ * @param router_module_name The name of the router module in the design for
+ *                           which the corresponding block id needs to be found.
+ * @param cluster_ctx Global variable that contains clustering information.
+ *                    Contains a datastructure to convert a module name to
+ *                    a cluster block id.
+ * @param single_flow_tag A xml tag that contains the traffic flow information
```

Comment:
mention which are just for error messages.

Reply:
done